### PR TITLE
Make shared bulk upload values consistent

### LINKED ISSUE #5456: Patient bulk upload file validations should match documentation
### Background
Noticed while testing the patient bulk uploader. Our documentation says `Unknown` is a valid value for `resident_congregate_setting` but the file validator will give an error if it's used. See screenshots.
Screenshot 2023-02-23 at 9 08 50 AM
Screenshot 2023-02-23 at 9 08 31 AM

### Action requested
Do an audit of our validation code and ensure we are allowing all values that are called out as allowed in our documentation.
Update code as necessary.
We should also be able to handle these values when actually creating the patients once the file is submitted.

### Acceptance Criteria
User can upload a file with any values specified in our documentation and the file should submit successfully and process correctly.

### Additional context
Be careful that changing the validations to accept certain values does not alter the test result bulk uploader to not match their documentation.

### PR DESCRIPTION
**Pull request by emyl3**
# BACKEND & FRONTEND PULL REQUEST

## Related Issue
- partially resolves #5456 
- #5456 also involves static site changes which will be made separately: [See here](https://github.com/CDCgov/prime-simplereport-site/pull/457) also deployed to dev2 for testing

## Changes Proposed
- This PR makes shared bulk upload values (across the test results bulk upload and patient bulk upload) like race, ethnicity, gender, employed in healthcare, and resident congregate setting accept the same values across both uploaders.

## Testing
- Deployed in dev2 for testing
- Verify alternate values for patient bulk upload works
- Ensure both bulk uploaders continue to work as intended

### CODE REVIEW

File: backend/src/test/java/gov/cdc/usds/simplereport/service/PatientBulkUploadServiceAsyncTest.java [hunk 1]
```diff
@@ -343,7 +386,7 @@ void personSaveAllFailure_triggersErrorEmail() throws IOException {
   }
 
   @Test
-  @SliceTestConfiguration.WithSimpleReportStandardUser
+  @SliceTestConfiguration.WithSimpleReportOtherOrgUser
```
Comment (emyl3) [inline backend/src/test/java/gov/cdc/usds/simplereport/service/PatientBulkUploadServiceAsyncTest.java:389]: ⚠️ This test kept failing with the addition of the new test above. The StandardUser in this configuration had the same email as the StandardAllFacilitiesUser so something weird was happening here. Just decided to make a new user in a different org to test this scenario.

File: frontend/src/app/testResults/uploads/__snapshots__/CsvSchemaDocumentation.test.tsx.snap [hunk 2]
```diff
@@ -1573,7 +1573,7 @@ exports[`CsvSchemaDocumentation tests CsvSchemaDocumentation matches snapshot 1`
                    or 
                   <mark>
                     <code>
-                      Ask, but unknown
```
[the remark this answers is not in the document]
Comment (BobanL) [reply frontend/src/app/testResults/uploads/__snapshots__/CsvSchemaDocumentation.test.tsx.snap:1576]: When I attempt to upload a file to Results uploader with `Ask but unknown` I get an error from Report Stream's validation. I believe (based on my CSV editor) that if you have a comma in a field you can just put the value in quotes and it seems to work like this `"Ask, but unknown"`.

File: backend/src/main/java/gov/cdc/usds/simplereport/validators/CsvValidatorUtils.java [hunk 3]
```diff
@@ -303,14 +326,22 @@ public static String convertEthnicityToDatabaseValue(String ethnicity) {
...
             Map.entry(WHITE_LITERAL, WHITE_LITERAL),
+            Map.entry(WHITE_CODE, WHITE_LITERAL),
             Map.entry(OTHER_LITERAL, OTHER_LITERAL),
-            Map.entry("ask but unknown", REFUSED_LITERAL),
-            Map.entry(UNKNOWN_LITERAL, UNKNOWN_LITERAL));
+            Map.entry(OTHER_RACE_CODE, OTHER_LITERAL),
+            Map.entry(ASK_LITERAL, REFUSED_LITERAL),
+            Map.entry(ASK_CODE, REFUSED_LITERAL),
+            Map.entry(UNKNOWN_LITERAL, UNKNOWN_LITERAL),
+            Map.entry(UNKNOWN_CODE, UNKNOWN_LITERAL));
```
Comment (BobanL) [inline backend/src/main/java/gov/cdc/usds/simplereport/validators/CsvValidatorUtils.java:344]: When I upload a [patient](https://dev2.simplereport.gov/app/patient/3c51c6d6-083d-4526-9080-bac45becef40?facility=1a0139b9-fc23-450b-9b6c-cd081e5cea9d) with "UNK" or "unknown", their race fields are missing when I look at it again. 
image

File: backend/src/main/java/gov/cdc/usds/simplereport/validators/CsvValidatorUtils.java [hunk 4]
```diff
@@ -292,9 +313,11 @@ public static MappingIterator<Map<String, String>> getIteratorForCsv(InputStream
...
         Map.ofEntries(
-            Map.entry("hispanic or latino", "hispanic"),
-            Map.entry("not hispanic or latino", "not_hispanic"),
-            Map.entry("unk", UNKNOWN_LITERAL),
+            Map.entry(HISPANIC_LITERAL, HISPANIC_DB_VALUE),
+            Map.entry(HISPANIC_CODE, HISPANIC_DB_VALUE),
+            Map.entry(NOT_HISPANIC_LITERAL, NOT_HISPANIC_DB_VALUE),
+            Map.entry(NOT_HISPANIC_CODE, NOT_HISPANIC_DB_VALUE),
+            Map.entry(UNKNOWN_CODE, UNKNOWN_LITERAL),
             Map.entry(UNKNOWN_LITERAL, UNKNOWN_LITERAL));
```
Comment (BobanL) [inline backend/src/main/java/gov/cdc/usds/simplereport/validators/CsvValidatorUtils.java:321]: Similar to Race, but when I upload a [patient](https://dev2.simplereport.gov/app/patient/e4476dbd-0191-48d3-8849-5d036ac3eeb3?facility=1a0139b9-fc23-450b-9b6c-cd081e5cea9d) with "UNK" or "UNKOWN" ethnicity that section is empty. 
image

**Review by BobanL — CHANGES_REQUESTED** [review]
It seems like "Unknown" codes don't map well back onto the UI when viewing the patients again. For both Ethnicity & Race the fields don't show up.

**Comment by emyl3** [comment]
> It seems like "Unknown" codes don't map well back onto the UI when viewing the patients again. For both Ethnicity & Race the fields don't show up.

Hmmm, so I was saving those as "unknown" in the database. I'm wondering if it makes sense to save the value for Race and Ethnicity "unknown" values to "refused" in the database like the gender field so they get marked as "Prefer not to answer" in the UI? 🤔

File: frontend/src/app/testResults/uploads/__snapshots__/CsvSchemaDocumentation.test.tsx.snap (hunk 2 above)
[the remark this answers is not in the document]
Comment (BobanL) [reply frontend/src/app/testResults/uploads/__snapshots__/CsvSchemaDocumentation.test.tsx.snap:1576]: This error comes from Report Stream, but if you look at [HL7 docs](https://hl7-definition.caristix.com/v2/HL7v2.7/Tables/0005) it shows asku isn't even valid for race 🤔

[the remark this answers is not in the document]
Comment (emyl3) [reply frontend/src/app/testResults/uploads/__snapshots__/CsvSchemaDocumentation.test.tsx.snap:1576]: OOO... I'll check-in with the rest of the team to see what this value was used for beforehand then 👀 Thank you for flagging that 😅

**Comment by emyl3** [comment]
> > It seems like "Unknown" codes don't map well back onto the UI when viewing the patients again. For both Ethnicity & Race the fields don't show up.
> 
> Hmmm, so I was saving those as "unknown" in the database. I'm wondering if it makes sense to save the value for Race and Ethnicity "unknown" values to "refused" in the database like the gender field so they get marked as "Prefer not to answer" in the UI? 🤔

From standup, it was decided to set these as "refused" so they display as "Prefer not to answer" in the UI

File: frontend/src/app/testResults/uploads/__snapshots__/CsvSchemaDocumentation.test.tsx.snap (hunk 2 above)
[the remark this answers is not in the document]
Comment (emyl3) [reply frontend/src/app/testResults/uploads/__snapshots__/CsvSchemaDocumentation.test.tsx.snap]: Changed it back to "Ask, but unknown" ReportStream lists it [as supported here](https://reportstream.cdc.gov/assets/pdf/ReportStream-Programmers-Guide-v3.0.pdf) but it throws an error when trying to submit that value. So @BobanL [flagged the issue here](https://skylight-hq.slack.com/archives/C01HFHPNJEL/p1679939568399459?thread_ts=1679927969.588359&cid=C01HFHPNJEL).

[the remark this answers is not in the document]
Comment (emyl3) [reply frontend/src/app/testResults/uploads/__snapshots__/CsvSchemaDocumentation.test.tsx.snap]: Will also need to update patient bulk upload guide PR for static site

**Comment by emyl3** [comment]
@BobanL ready for re-review! Thank you!!

**Review by BobanL — APPROVED** [review]
Changes look good to me! Thank you!!

**Review by johanna-skylight — APPROVED** [review]
LGTM!

**Pull request merged**